configure.ac

```diff
@@ -338,23 +364,24 @@ if test "x$PRESET_CFLAGS" = "x"; then
 fi
 
 dnl set squid required flags
+dnl TODO use SQUID_CC_ADD_CXXFLAG_IF_SUPPORTED instead of hardcoding
```

Comment:
Would you mind if I deferred this to a followup PR once testing on mingw is possible?

Reply:
I'd rather not. Most of these details are for GCC-3. mingw is currently quite broken and when that testing is done it can do any mingw-specific flags back in.

Reply:
Done. It blossomed into a massive 95 files change. In the process I discovered stubs having mismatched prototypes, unnecessary stubs, and a bunch of other tidbits